crates/sel4-capdl-initializer/core/src/lib.rs

```diff
@@ -326,6 +326,23 @@ impl<'a, N: ObjectName, D: Content, M: GetEmbeddedFrame, B: BorrowMut<[PerObject
             }
         }
 
+        sel4::sel4_cfg_if! {
+            if #[sel4_cfg(ARCH_X86_64)] {
+                let ioports = self
+                    .spec()
+                    .filter_objects::<&object::IOPorts>()
+                    .map(|(obj_id, obj)| (obj_id, obj.start_port, obj.end_port));
+
+                for (obj_id, start_port, end_port) in ioports {
+                    let slot = self.cslot_alloc_or_panic();
+                    init_thread::slot::IO_PORT_CONTROL
+                        .cap()
+                        .ioport_control_issue(start_port, end_port, &cslot_to_relative_cptr(slot))?;
+                    self.set_orig_cslot(obj_id, slot);
+                }
+            }
+        }
+
```

Comment:
Could you try this sort of conditional block instead?

```
#[sel4_cfg(ARCH_X86_64)]
{
  ...
}
```

If the compiler doesn't accept that, then could you do this, to wrap the whole thing in a new scope?

```
sel4::sel4_cfg_if! {
    if #[sel4_cfg(ARCH_X86_64)] {
        {
            let ...
        }
    }
}
```

Reply:
Also, could you move this block below the IrqHandler block beneath it?